p2p/dialer.go

```diff
@@ -35,7 +41,7 @@ import (
 type Dialer interface {
 	// Dial exchanges p2p configuration via broker, performs NAT pinging if needed
 	// and create p2p channel which is ready for communication.
-	Dial(consumerID, providerID identity.Identity, serviceType string, timeout time.Duration) (Channel, error)
+	Dial(consumerID, providerID identity.Identity, serviceType string, ctx context.Context) (Channel, error)
```

Comment:
It's a good practice to always pass context as a first parameter.

```go
Dial(ctx context.Context, consumerID, providerID identity.Identity, serviceType string) (Channel, error)
```

You can also change provider listener to use context 👍

Reply:
Lets implement that in separate PR